core/src/main/java/io/fabric8/maven/core/util/KubernetesResourceUtil.java

```diff
@@ -719,9 +719,20 @@ public static void mergePodSpec(PodSpecBuilder builder, PodSpec defaultPodSpec,
                 int idx = 0;
                 for (Container defaultContainer : defaultContainers) {
                     Container container;
+                    boolean isNewContainerAdded = false;
                     if (idx < containers.size()) {
                         container = containers.get(idx);
                     } else {
+                        isNewContainerAdded = true;
+                        container = new Container();
+                        containers.add(container);
+                    }
+                    //TODO: SK remove it after review
+                    //Though we have more containers in the builder, but if existing one does not match
+                    // default container name then we assume it to be new one or the one to be added
+                    String containerName = container.getImage();
+                    String defaultContainerName = defaultContainer.getImage();
+                    if (!containerName.equalsIgnoreCase(defaultContainerName) && !isNewContainerAdded) {
```

Comment:
Here you're correlating over the image name. This works with sidecars, but it may have problems with pods having multiple containers from the same base image (with different startup options).

I think a better approach would be to correlate over the container image, i.e.
- If a fragment has a name, and the name is different the one that would be used for the current project, then add another container
- If the fragment does not declare a name, or the name matches the one that would be used for the current project, then use that as container for the project